presto-main/src/main/java/com/facebook/presto/dispatcher/LocalDispatchQuery.java

```diff
@@ -97,6 +104,63 @@ public LocalDispatchQuery(
         });
     }
 
+    @Override
+    public void startWaitingForPrerequisites()
+    {
+        // It's possible that queryExecution fails before we start for prerequisites, in that case, don't even
+        // start waiting for prerequisites
+        if (isDone()) {
+            return;
+        }
+
+        try {
+            Session session = stateMachine.getSession();
+            CompletableFuture<?> prerequisitesFuture = queryPrerequisites.waitForPrerequisites(
+                    stateMachine.getQueryId(),
+                    new QueryPrerequisitesContext(
+                            session.getCatalog(),
+                            session.getSchema(),
+                            stateMachine.getBasicQueryInfo(Optional.empty()).getQuery(),
+                            session.getSystemProperties(),
+                            session.getConnectorProperties()));
+
+            addStateChangeListener(state -> {
+                if (state.isDone()) {
+                    queryPrerequisites.queryFinished(stateMachine.getQueryId());
+                    if (!prerequisitesFuture.isDone()) {
+                        prerequisitesFuture.cancel(true);
+                    }
+                }
+            });
+
+            prerequisitesFuture.whenCompleteAsync((result, throwable) -> {
+                if (throwable != null) {
+                    fail(throwable);
+                    return;
+                }
+
+                queueQuery();
+            }, queryExecutor);
+        }
+        catch (Throwable t) {
+            fail(t);
+            log.warn(t, "Query %s failed while submitting for prerequisite check", stateMachine.getQueryId());
```

Comment:
log-and-throw anti-pattern?

Reply:
I was following what has already been done in this class - but I agree - we can get rid of it